### DISCUSSION TITLE
Any plan to move to pytorchlightning 2.0

### DISCUSSION THREAD

**Discussion by YuyangXueEd** [facebookresearch/fastMRI#307] [General]
Hello!

Since PyTorch and PytorchLightning is upgraded to 2.0 and above, the `fastmri_examples` code such as `unet` is now deprecated. I am wondering that whether there is a plan to update the code to compatible to 2.0.

There are some changes like:
- used `LightningDataModule.parse_argparser()`, `LightningDataModule.from_argparse_args()` and `Trainer.from_argparse_args()` method now switch to using `LightningCLI`.
- had any logic except reducing the DP outputs in `LightningModule.validation_step_end` hook now port it to `LightningModule.on_validation_batch_end` hook
- had any logic except reducing the DP outputs in `LightningModule.test_step_end` hook now port it to `LightningModule.on_test_batch_end` hook
- no `resume_from_checkpoint` for the `Trainer` but changed to `ckpt_path`
- ...

**Comment by mmuckley** [comment]
Hello @YuyangXueEd, I'm spending my time on compression right now so I'm unable to make repository updates. Feel free to open an Issue with the Enhancement tag for this - maybe someone will test out all the changes and open a pull request.

**Reply by YuyangXueEd** [reply]
Thank you for your reply!

I am now implementing UNet using a newer version of `pl 2.0.2` and `torch 2.0.1`. I will try to do a pull request when things done smoothly.

Best!
